src/rastervision/tagging/tasks/predict.py

```diff
@@ -101,40 +101,82 @@ def compute_probs(run_path, model, options, generator, split):
                 flip_img = np.fliplr(img)
                 aug_batch.append(np.expand_dims(flip_img, axis=0))
                 aug_batch = np.concatenate(aug_batch, axis=0)
-
-                aug_probs = np.mean(model.predict(aug_batch), axis=0,
-                                    keepdims=True)
-
-                y_probs.append(aug_probs)
-        # Otherwise, predicting probs on unaugmented images by batch
-        else:
+                aug_probs = model.predict(aug_batch)
+
+                for aug_ind in range(NUM_AUG):
+                    y_probs[aug_ind].append(aug_probs[aug_ind])
+
+            if (options.nb_eval_samples is not None and
+                    batch_ind * options.batch_size >= options.nb_eval_samples):
+                break
+
+        for aug_ind in range(NUM_AUG):
+            y_probs[aug_ind] = np.concatenate([y_probs[aug_ind]])
```

Comment:
Intuitively I would agree with you, but I actually added the outer list to get concatenate to work properly. If it were just a list of numpy arrays, concatenating put all of the values into one giant numpy array and lost all information of individual samples. Putting it inside another list is what kept the dimensions since np.concatenate doesn't have a keepdims arg.